### ISSUE TITLE
Please bump ruby-saml to v1.0.0 to get signed metadata.xml support

### ISSUE THREAD

**Issue by dkelson** [PracticallyGreen/omniauth-saml#47]
I'm using omniauth-saml as part of GitLab and trying to connect to a a ipsilon idP that is backed by FreeIPA. The ipsilon idP requires that the metadata be signed. The ruby-saml-0.8.2 that is currently used in omniauth-saml doesn't do signed metadata.

According to the ruby-saml changelog signed meta-data support was added in v0.9.2. The current version is 1.0.0. The rub-saml website says, "Version 1.0 is a recommended update for all Ruby SAML users as it includes security fixes."

FYI, here is the configuration I'm trying to use.

```
gitlab_rails['omniauth_providers'] = [
   {
     "name" => "saml",
     "args" => {
         assertion_consumer_service_url:  'https://code.example.com/users/auth/saml/callback',
         idp_cert_fingerprint: 'B7:4E:15:77:40:DE:EE:EB:77:53:EE:58:4E:87:2B:3F:6B:8A:64:87',
         idp_sso_target_url: 'https://idp.example.com/idp/saml2/SSO/Redirect',
         issuer: 'GLGitLab',
         name_identifier_format: 'urn:oasis:names:tc:SAML:2.0:nameid-format:transient',
         certificate: '-----BEGIN CERTIFICATE-----
<snipped_for_brevity>
-----END CERTIFICATE-----',
        private_key: '-----BEGIN PRIVATE KEY-----
<snipped_for_brevity>
-----END PRIVATE KEY-----',
        "security" => {
                authn_requests_signed: true,
                metadata_signed: true,
                digest_method: 'XMLSecurity::Document::SHA1',
                signature_method: 'XMLSecurity::Document::RSA_SHA1',
                embed_sign: true
                }
        }
   }
 ]
```

**Comment by nel** [comment]
+1 they are reported vulnerabilies in ruby-saml 0.8.2- and people using omniauth-saml cannot upgrade due to omniauth-ruby dep on ruby-saml ~> 0.8.1

    Name: ruby-saml
    Version: 0.8.2
    Advisory: OSVDB-124383
    Criticality: Medium
    URL: https://github.com/onelogin/ruby-saml/pull/247
    Title: Ruby-Saml Gem is vulnerable to entity expansion attacks
    Solution: upgrade to >= 1.0.0

    Name: ruby-saml
    Version: 0.8.2
    Advisory: OSVDB-124991
    Criticality: High
    URL: https://github.com/onelogin/ruby-saml/pull/225
    Title: Ruby-Saml Gem is vulnerable to XPath Injection
    Solution: upgrade to >= 1.0.0

**Comment by booleanbetrayal** [comment]
I've submitted a PR #46 for this ... please help in bumping priority there. wondering if this repo needs a new owner ...

**Comment by bpedro** [comment]
I just released v1.4.0 (https://github.com/PracticallyGreen/omniauth-saml/releases/tag/v1.4.0) which is using ruby-saml v1.0.0.
